File: StudentPortfolio.Client/src/types/dtos/student.ts
import type { BaseEntity } from "./_baseEntity";
import type { BaseResponse } from "./_baseResponse";
import type { Acknowledgement } from "./acknowledgement";

export interface Student extends BaseEntity {
  id: string;
  institutionalId: string;
  name: string;
  lastName: string;
  fullName: string;
  startDate: string;
  endDate: string;
  institution: string;
  acknowledgements?: Acknowledgement[];
}

export type MutateStudentRequest = Partial<
  Omit<Student, "Id" | "acknowledgements" | "fullName">
>;

export type CreateStudentRequest = MutateStudentRequest;
export type UpdateAcknowledgementRequest = MutateStudentRequest;

export type GetStudentResponse = BaseResponse<Student>;
export type GetStudentsResponse = BaseResponse<Student[]>;
export type GetStudentRequest = Record<string, string>;
export type UpdateStudentRequest = CreateStudentRequest;

File: StudentPortfolio/Models/Entities/Student.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using StudentPortfolio.Models.Infrastructure;
using System.ComponentModel.DataAnnotations.Schema;

namespace StudentPortfolio.Models.Entities
{
    public class Student : IModel, IDeletable
    {
        public Guid Id { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string InstitutionalId { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string Name { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string LastName { get; set; }
        public DateOnly? StartDate { get; set; }
        public DateOnly? EndDate { get; set; }

        [Column(TypeName = "NVARCHAR(255)")]
        public string Institution { get; set; }
        public bool Deleted { get; set; }
        public DateTimeOffset? DeletedAt { get; set; }
        public DateTimeOffset DateCreated { get; set; }
        public ICollection<Acknowledgement> Acknowledgements { get; set; }
    }
}
